pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -0,0 +1,1438 @@
+		createProvisionedPVInterval:   createProvisionedPVInterval,
+	}
+	controller.recyclePluginMgr.InitPlugins(recyclers, controller)
+	if err := controller.provisioner.Init(controller); err != nil {
+		glog.Errorf("PersistentVolumeController: error initializing provisioner plugin: %v", err)
+	}
+
+	volumeSource := &cache.ListWatch{
+		ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+			return kubeClient.Core().PersistentVolumes().List(options)
+		},
+		WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+			return kubeClient.Core().PersistentVolumes().Watch(options)
+		},
+	}
+
+	claimSource := &cache.ListWatch{
+		ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+			return kubeClient.Core().PersistentVolumeClaims(api.NamespaceAll).List(options)
+		},
+		WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+			return kubeClient.Core().PersistentVolumeClaims(api.NamespaceAll).Watch(options)
+		},
+	}
+
+	controller.initializeController(syncPeriod, volumeSource, claimSource)
+
+	return controller
+}
+
+func (ctrl *PersistentVolumeController) initializeController(syncPeriod time.Duration, volumeSource, claimSource cache.ListerWatcher) {
+	glog.V(5).Infof("Initializing PersistentVolumeController, sync every %s", syncPeriod.String())
+	ctrl.volumes.store, ctrl.volumeController = framework.NewIndexerInformer(
+		volumeSource,
+		&api.PersistentVolume{},
+		syncPeriod,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    ctrl.addVolume,
+			UpdateFunc: ctrl.updateVolume,
+			DeleteFunc: ctrl.deleteVolume,
+		},
+		cache.Indexers{"accessmodes": accessModesIndexFunc},
+	)
+	ctrl.claims, ctrl.claimController = framework.NewInformer(
+		claimSource,
+		&api.PersistentVolumeClaim{},
+		syncPeriod,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    ctrl.addClaim,
+			UpdateFunc: ctrl.updateClaim,
+			DeleteFunc: ctrl.deleteClaim,
+		},
+	)
+}
+
+func (ctrl *PersistentVolumeController) addVolume(obj interface{}) {
+	if !ctrl.isFullySynced() {
+		return
+	}
+
+	pv, ok := obj.(*api.PersistentVolume)
+	if !ok {
+		glog.Errorf("Expected PersistentVolume but handler received %+v", obj)
+		return
+	}
+	if err := ctrl.syncVolume(pv); err != nil {
+		glog.Errorf("PersistentVolumeController could not add volume %q: %+v", pv.Name, err)
+	}
+}
+
+func (ctrl *PersistentVolumeController) updateVolume(oldObj, newObj interface{}) {
+	if !ctrl.isFullySynced() {
+		return
+	}
+
+	newVolume, ok := newObj.(*api.PersistentVolume)
+	if !ok {
+		glog.Errorf("Expected PersistentVolume but handler received %+v", newObj)
+		return
+	}
+	if err := ctrl.syncVolume(newVolume); err != nil {
+		glog.Errorf("PersistentVolumeController could not update volume %q: %+v", newVolume.Name, err)
+	}
+}
+
+func (ctrl *PersistentVolumeController) deleteVolume(obj interface{}) {
+	if !ctrl.isFullySynced() {
+		return
+	}
+
+	var claim *api.PersistentVolumeClaim
+	if volume, ok := obj.(*api.PersistentVolume); ok {
```

Comment:
You're dropping the case where this conversion fails; you can break it out and save an indent level for the rest of this block.